### PR TITLE
acdc view change to reduce the size

### PR DESCRIPTION
**Pull request by ticoann**
#5567 need to be updated before this is applied to agents

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, @vkuznet, please review.
1. two problematic view are removed (#5567)
2. all the sorting and chunking is done in client side (might be memory intensive).
3. unit test is passed.

Although unittest is passed, I am not sure there might be something wrong. There are a few things I would like to fix (location format is different in different method, create the cache so it won't make multiple calls when chucking is applied) but I just want to make it work (pass the same unit test initially).

File: src/python/WMCore/ACDC/DataCollectionService.py [hunk 1]
```diff
@@ -82,6 +82,37 @@ def failedJobs(self, failedJobs, useMask = True):
...
+        
+        filesInfo = filesetFiles.values()
+        
+        # second lfn sort
+        filesInfo.sort(key = lambda x: x["lfn"])
+        #primary location sort (python preserve sort result) 
+        filesInfo.sort(key = lambda x: "".join(self._sortLocationInPlace(x)))
+        
+        if chunkOffset != None and chunkSize != None:
+            return filesInfo[chunkOffset: chunkOffset + chunkSize]
```
Comment (amaltaro) [inline src/python/WMCore/ACDC/DataCollectionService.py]: Is it a typo? Should it be [chunkOffset + chunkSize] instead?

**Comment by amaltaro** [comment]
I cannot review it. This view is empty in both testbed and production dbs:
https://cmsweb-testbed.cern.ch/couchdb/acdcserver/_design/ACDC/_view/owner_coll_fileset_docs
so I don't know which info is being retrieved :( (unless I'm doing something stupid)

**Comment by vkuznet** [comment]
Changes seem reasonable to me. Alan, the syntax filesInfo[chunkOffset: chunkOffset + chunkSize] seems to be correct since it returns a range of elements between chunkOffset and its position plus chunk size.

**Comment by amaltaro** [comment]
Ahaaa!! That's right Valentin. Thanks

**Comment by ticoann** [comment]
Hi guys, Thank you very much for prompt review.

> I cannot review it. This view is empty in both testbed and production dbs:
> https://cmsweb-testbed.cern.ch/couchdb/acdcserver/_design/ACDC/_view/owner_coll_fileset_docs

Actually we have to add options to the call.
https://cmsweb-testbed.cern.ch/couchdb/acdcserver/_design/ACDC/_view/owner_coll_fileset_docs?reduce=false&include_docs=true

Alan, we do normally do ACDC testing for integration test, right?

**Comment by amaltaro** [comment]
Thanks for the url Seangchan!
Yes, we usually have ACDC workflows, but in this case we need to evaluate all the bits. I'll send you another mail

**Comment by ticoann** [comment]
Thanks Alan, To test this, we need update agent with #5567

**Comment by ticoann** [comment]
This need to be updated to global workqueue as well

**Pull request merged**